### PR TITLE
Update @balena/lint to 7.0.0

### PR DESCRIPTION
**Pull request by thgreasi**
Update @balena/lint from 6.2.2 to 7.0.0

Depends-on: https://github.com/balena-io-modules/node-balena-lint/pull/85
Depends-on: https://github.com/balena-io-modules/abstract-sql-to-typescript/pull/40
Change-type: patch

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
 	"include": [
 		"src/**/*.ts",
 		"typings/**/*.d.ts",
+		"*.js",
```
Comment (Page-) [inline tsconfig.json:9]: Why is this necessary?

File: test/13_loki-backend.ts [hunk 2]
```diff
@@ -111,7 +111,7 @@ describe('loki backend', () => {
 		const loki = new LokiBackend();
 		await new Bluebird(async (resolve) => {
 			let countLogs = 0;
-			loki.subscribe(ctx, () => {
+			await loki.subscribe(ctx, () => {
```
Comment (Page-) [inline test/13_loki-backend.ts:114]: I feel like this might affect the timings of things

File: src/features/device-state/routes/state-get-v2.ts [hunk 3]
```diff
@@ -58,8 +58,8 @@ export type StateV2 = {
 		};
 	};
 	dependent: {
-		apps: {};
-		devices: {};
+		apps: object;
+		devices: object;
```
Comment (Page-) [inline src/features/device-state/routes/state-get-v2.ts:62]: Fwiw I think these should be specifically an empty object, I don't know off the top of my head how you type that?

File: src/features/device-proxy/device-proxy.ts [hunk 4]
```diff
@@ -132,10 +132,8 @@ export const proxy = async (req: Request, res: Response) => {
 		if (handleHttpErrors(req, res, err)) {
 			return;
 		}
-		if (err?.body != null) {
-			err = err.body;
-		}
-		res.status(502).send(translateError(err));
+		const errorToReturn = err?.body != null ? err.body : err;
```
Comment (Page-) [inline src/features/device-proxy/device-proxy.ts:135]: Am I right in thinking this can be
```suggestion
		const errorToReturn = err?.body?? err;
```
?

File: test/13_loki-backend.ts (hunk 2 above)
Comment (thgreasi) [reply test/13_loki-backend.ts:114]: I though that given that we already have a `setTimeout(100)` right below to "wait for a bit" it would be fine, but I'm fine keeping it as is for this PR :+1:

File: tsconfig.json (hunk 1 above)
Comment (thgreasi) [reply tsconfig.json:9]: The TypeScript-ESLint plugin can only lint files that are part of the tsconfig.json (or any other tsconfig.something.json that we might be pointing it to), otherwise it errors with:
```
error  Parsing error: ESLint was configured to run on `<tsconfigRootDir>/index.js` using `parserOptions.project`: <tsconfigRootDir>/tsconfig.json
However, that TSConfig does not include this file. Either:
- Change ESLint's list of included files to not include this file
- Change that TSConfig to include this file
- Create a new TSConfig that includes this file and include it in your parserOptions.project
See the typescript-eslint docs for more info: https://typescript-eslint.io/linting/troubleshooting#i-get-errors-telling-me-eslint-was-configured-to-run--however-that-tsconfig-does-not--none-of-those-tsconfigs-include-this-file
```

File: test/13_loki-backend.ts [hunk 5]
```diff
@@ -99,6 +99,7 @@ describe('loki backend', () => {
 		const loki = new LokiBackend();
 		const log = createLog();
 		const incomingLog = await new Bluebird(async (resolve) => {
+			// eslint-disable-next-line @typescript-eslint/no-floating-promises
 			loki.subscribe(ctx, resolve);
```
Comment (fisehara) [inline test/13_loki-backend.ts:103]: Why woudln't await make sense here? Would it even make the `await setTimeout(100)` obsolete?

File: test/13_loki-backend.ts [hunk 6]
```diff
@@ -111,6 +112,7 @@ describe('loki backend', () => {
 		const loki = new LokiBackend();
 		await new Bluebird(async (resolve) => {
 			let countLogs = 0;
+			// eslint-disable-next-line @typescript-eslint/no-floating-promises
```
Comment (fisehara) [inline test/13_loki-backend.ts:115]: See above

File: init.ts [hunk 7]
```diff
@@ -138,6 +138,7 @@ async function onInitHooks() {
 	});
 
 	// this will pre-fetch the device types and populate the cache...
+	// eslint-disable-next-line @typescript-eslint/no-floating-promises
 	getAccessibleDeviceTypes(sbvrUtils.api.resin);
```
Comment (fisehara) [inline init.ts:142]: Wouldn't awaiting here create a guarantee that we may need?

**Review by fisehara — COMMENTED** [review]
LGTM 
NIT two `await import` candiates.
NIT two await candidates

File: test/13_loki-backend.ts (hunk 5 above)
Comment (thgreasi) [reply test/13_loki-backend.ts:103]: I actually had this await originally but @Page- was concerned about it affecting the timings, so I decided to keep this a no-op.
See: https://github.com/balena-io/open-balena-api/pull/1396#discussion_r1301675512

File: init.ts (hunk 7 above)
Comment (thgreasi) [reply init.ts:142]: We don't need to await this, since slow or non-accessible S3 shouldn't block the API startup.
As per the pre-existing comment `// this will pre-fetch the device types and populate the cache...` this is here only to warm-up the device-type.json cache.

File: src/index.ts [hunk 8]
```diff
@@ -478,6 +477,7 @@ async function runCommand(
 	cmd: string,
 	argv: string[],
 ): Promise<void> {
+	// eslint-disable-next-line @typescript-eslint/no-var-requires
 	const script = require(path.join(__dirname, 'commands', cmd));
```
[the remark this answers is not in the document]
Comment (thgreasi) [reply src/index.ts:481]: It seems that we no longer use this function and we can drop it since there is only a dymmy command atm
See: https://github.com/balena-io/open-balena-api/pull/22/files

File: init.ts (hunk 7 above)
Comment (thgreasi) [reply init.ts:142]: Let me update the comment

**Review by fisehara — APPROVED** [review]
